core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -2286,6 +2301,13 @@ public RelBuilder sortLimit(int offset, int fetch, RexNode... nodes) {
     return sortLimit(offset, fetch, ImmutableList.copyOf(nodes));
   }
 
+  /** Creates a {@link Sort} by collation, with limit and offset. */
+  public RelBuilder sortLimit(RelCollation collation, RexNode offset, RexNode fetch) {
+    final RelNode sort = sortFactory.createSort(peek(), collation, offset, fetch);
+    replaceTop(sort);
+    return this;
+  }
+
```

Comment:
This method is not necessary. Instead of `builder.sortLimit(collation, offset, fetch)` you can call `builder.sortLimit(offset, fetch, builder.fields(collation))`.

Reply:
Done. Thanks for the suggestion.